src/main/java/com/fullcontact/api/libs/fullcontact4j/http/cardreader/CardReaderUploadRequest.java

```diff
@@ -1,56 +1,97 @@
+import java.io.File;
 import java.io.InputStream;
 import java.util.Map;
 
 public class CardReaderUploadRequest extends FCRequest<CardReaderUploadConfirmResponse> {
 
-    private RequestBodyJson body;
+    private String cardFrontStr;
+    private String cardBackStr;
+    private File cardFrontFile;
+    private File cardBackFile;
+    private String mimeType;
     private String accessToken;
 
-    protected CardReaderUploadRequest(String accessToken, String front, String back, Map<String, String> params) {
+    public CardReaderUploadRequest(Map<String, String> params, String cardFrontStr, String cardBackStr, File
+        cardFrontFile, File cardBackFile, String mimeType, String accessToken) {
         super(params);
+        this.cardFrontStr = cardFrontStr;
+        this.cardBackStr = cardBackStr;
+        this.cardFrontFile = cardFrontFile;
+        this.cardBackFile = cardBackFile;
+        this.mimeType = mimeType;
         this.accessToken = accessToken;
-        body = new RequestBodyJson();
-        body.setFront(front);
-        body.setBack(back);
     }
 
     @Override
     protected void makeRequest(FullContactApi api, Callback<CardReaderUploadConfirmResponse> callback) {
-       //if accesstoken is null, it is not added to the headers. if it is not null, it replaces API Key header
-       //(see FullContactHttpInterface.DynamicHeaderOkClient)
-       api.uploadCard(accessToken, params, body, callback);
+        //if accesstoken is null, it is not added to the headers. if it is not null, it replaces API Key header
+        //(see FullContactHttpInterface.DynamicHeaderOkClient)
+        if(cardFrontStr != null) {
+            RequestBodyJson body = new RequestBodyJson();
+            body.setFront(cardFrontStr);
+            body.setBack(cardBackStr);
+            api.uploadCard(accessToken, params, body, callback);
+        } else {
+            if(cardBackFile == null) {
+                api.uploadCardForm(accessToken, params, new TypedFile(mimeType, cardFrontFile), callback);
+            } else {
+                api.uploadCardForm(accessToken, params, new TypedFile(mimeType, cardFrontFile),
+                    new TypedFile(mimeType, cardBackFile), callback);
+            }
+        }
     }
 
     public static class Builder extends WebhookBuilder<Builder, CardReaderUploadRequest> {
 
-        private String cardFront;
+        private String cardFrontStr;
+        private String cardBackStr;
+        private File cardFrontFile;
+        private File cardBackFile;
+        private String mimeType;
         private String accessToken;
-        private String cardBack;
 
         public Builder verified(CardReaderQuality quality) {
             params.put(FCConstants.PARAM_CARD_VERIFIED, quality.name());
             return this;
         }
 
         public Builder cardFront(InputStream stream) {
-            cardFront = Utils.encodeToStringAndClose(stream);
+            cardFrontStr = Utils.encodeToStringAndClose(stream);
```

Comment:
encodeToStringAndClose is the method I was originally concerned with.  This turns the stream into a string in memory.   I'm hoping there's a way to send the stream through Retrofit without turning it into a String.  The String inhales the file into memory which is a potential scaling problem.  Really I'd like the library to work with the stream throughout.  Wanted cardFront to take an InputStream which is then streamed through Retrofit and onto FullContact - all without inhaling stream into a String in memory.